frontend/app/src/App.tsx

```diff
@@ -580,20 +593,34 @@ export class App extends PureComponent<Props, State> {
       `Connection state changed from ${this.state.connectionState} to ${newState}`
     )
 
-    this.setState({ connectionState: newState })
-
     if (newState === ConnectionState.CONNECTED) {
       logMessage("Reconnected to server; requesting a script run")
       // Trigger a full app rerun:
       this.widgetMgr.sendUpdateWidgetsMessage(undefined)
       this.setState({ dialog: null })
+
+      this.hostCommunicationMgr.sendMessageToHost({
+        type: "WEBSOCKET_CONNECTED",
+      })
     } else {
+      // If we're starting from the CONNECTED state and going to any other
+      // state, we must be disconnecting.
+      if (this.state.connectionState === ConnectionState.CONNECTED) {
+        this.hostCommunicationMgr.sendMessageToHost({
+          type: "WEBSOCKET_DISCONNECTED",
```

Comment:
It's possible that we might want to send this information eventually, but I think it'd be slightly preferable to not add it for now while it won't be used on the SiS side to prevent our number of message types from growing unnecessarily.